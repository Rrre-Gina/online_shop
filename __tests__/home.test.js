import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../pages/index';
import * as nextRouter from 'next/router';

nextRouter.useRouter = jest.fn();
nextRouter.useRouter.mockImplementation(() => ({ route: '/' }));

it("render Home page with heading", () => {
  	render(<Home />);
	expect(screen.getByRole("heading")).toHaveTextContent('Home');
	const tree = renderer.create(<Home />).toJSON();
	expect(tree).toMatchSnapshot();
});