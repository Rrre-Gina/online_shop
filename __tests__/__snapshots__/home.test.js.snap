// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home page with heading 1`] = `
[
  <nav>
    <li>
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Главная
      </a>
    </li>
    <li>
      <a
        href="/categories"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Категории
      </a>
    </li>
    <li>
      <a
        href="/cart"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Корзина
      </a>
    </li>
    <li
      className="dropdown"
    >
      <a
        href="/profile"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Профиль
      </a>
      <div
        className="dropdown-content"
      >
        <a
          href="/profile"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Профиль
        </a>
        <a
          href="/auth"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Выход
        </a>
      </div>
    </li>
  </nav>,
  <main>
    <h2>
      Home
    </h2>
  </main>,
]
`;
